### PR TITLE
[Dashing] Adds CMAKE_CXX_FLAGS to extra_cmake_args

### PR DESCRIPTION
**Pull request by nuclearsandwich**
Backport of #28 for Dashing.

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13126/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8057/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10835/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13119/)

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,7 +22,7 @@ if(NOT Poco_FOUND)
   if(WIN32)
     list(APPEND POCO_CXX_FLAGS "/wd4244 /wd4530 /wd4577")
   else()
-    list(APPEND extra_cmake_args "-DCMAKE_C_FLAGS=-Wno-shift-negative-value")
+    list(APPEND extra_cmake_args "-DCMAKE_C_FLAGS=${CMAKE_C_FLAGS} -Wno-shift-negative-value")
```
Comment (gbiggs) [inline CMakeLists.txt:25]: Is this meant to be `CMAKE_CXX_FLAGS`?

Comment (nuclearsandwich) [reply CMakeLists.txt:25]: Possibly but it was reviewed and merged verbatim in #28

Comment (gbiggs) [reply CMakeLists.txt:25]: OK, just checking because the PR title says CXX.

**Review by gbiggs — APPROVED** [review]

**Pull request merged**